test/models/report_test.rb

```diff
@@ -0,0 +1,36 @@
+require "test_helper"
+
+class ReportTest < ActionController::TestCase
+  test "valid? returns false when report is too short" do
+    msg = "Report is too short (minimum is 100 characters)"
+    report = Report.new
+
+    assert_not report.valid?
+    assert_equal msg, report.errors.full_messages.first
+  end
+
+  test "valid? returns false when report is too big" do
+    msg = "Report is too long (maximum is 20000 characters)"
+    data = (1..20_000).map { rand(10000).to_s }.join
+    report = Report.new(report: data)
+
+    assert_not report.valid?
+    assert_equal msg, report.errors.full_messages.first
```

Comment:
We could do a more generic assertion like `assert_includes report.errors.full_messages, msg`, that way the test does not depend on the order of the error but only on having the specific error present.